auto-open issues for failed tests

tbg opened the pull request:
this is a little experiment to make it easier to deal with test failures on the master branch.
using issues, it will be easier to assign to fixing a particular build.

unfortunately, CircleCI as of now has no way of checking via circle.yml what the test status is (unless we add a hack for each test, which is clumsy), so we grep the log files for interesting info and open issues whenever something that looks like a test failure is found.

until we set a GITHUB_API_TOKEN environment variable, this will not create issues. Once we do, the result will look something like this:

https://github.com/tschottdorf/cockroach/issues/11

bdarnell commented on circle.yml:
```diff
@@ -46,6 +46,19 @@ test:
     - "[ -f ${CIRCLE_ARTIFACTS}/testrace.log ] && docker run -i cockroachdb/cockroach-dev shell /go/bin/go2xunit < ${CIRCLE_ARTIFACTS}/testrace.log > ${CIRCLE_TEST_REPORTS}/race/testrace.xml"
+    - |
+      find "${CIRCLE_ARTIFACTS}" -name '*.log' -type f -exec \
+        grep -B 5 -A 10 -E '\-? *FAIL|^panic|^[Gg]oroutine \d+|(read|write) by.*goroutine|DATA RACE' {} ';' \
```
The `\-? *` at the start of the regex looks redundant. 

It's sad that even after parsing the logs into a structured format with go2xunit, it's still easier to grep than to parse the xml.

bdarnell commented:
LGTM

tbg committed:
auto-open issues for failed tests

unfortunately, CircleCI as of now has no way of checking
via circle.yml what the test status is (unless we add a
hack for each test, which is clumsy), so we grep the
log files for interesting info and open issues whenever
something that looks like a test failure is found.

using issues, it will be easier to assign to fixing a par-
ticular build.
